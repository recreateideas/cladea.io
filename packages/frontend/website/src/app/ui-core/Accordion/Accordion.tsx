import { ReactElement, useState, useEffect } from 'react';
import { ExpandMore as ExpandMoreIcon } from '@mui/icons-material';
import {
    Container,
    Accordion as StyledAccordion,
    AccordionSummary,
    Title,
    AccordionDetails,
} from './styledComponents';

interface IProps {
    onChange?: (...args: any) => void;
    expanded?: boolean;
    controlled?: boolean;
    children?: ReactElement | Array<ReactElement>;
    title?: ReactElement | string;
    className?: string;
    expandIcon?: ReactElement;
    defaultExpanded?: boolean;
}
export const Accordion = (props: IProps): ReactElement => {
    const {
        children,
        controlled = false,
        expanded = false,
        title,
        className = '',
        expandIcon = <ExpandMoreIcon />,
        defaultExpanded = false,
    } = props;
    const [isExpanded, setIsExpanded] = useState<boolean>(defaultExpanded);

    const onChange = () => {
        setIsExpanded((exp) => !exp);
    };
    useEffect(() => {
        if (controlled && isExpanded !== expanded) {
            setIsExpanded(!!expanded);
        }
    }, [controlled, isExpanded, expanded]);
    return (
        <Container className={`${className}`}>
            <StyledAccordion
                {...{
                    TransitionProps: { unmountOnExit: true },
                    expanded: isExpanded,
                    onChange,
                }}
            >
                <AccordionSummary
                    className="accordion-summary"
                    {...{
                        expandIcon,
                    }}
                >
                    <Title>{title}</Title>
                </AccordionSummary>
                <AccordionDetails className="accordion-details">{children}</AccordionDetails>
            </StyledAccordion>
        </Container>
    );
};

Accordion.displayName = 'Accordion';

Accordion.defaultProps = {};
