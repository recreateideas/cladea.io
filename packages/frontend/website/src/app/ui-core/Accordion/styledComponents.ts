import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
