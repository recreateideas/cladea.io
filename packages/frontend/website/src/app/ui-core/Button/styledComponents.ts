import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
