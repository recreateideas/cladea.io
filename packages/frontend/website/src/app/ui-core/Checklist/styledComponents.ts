import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
