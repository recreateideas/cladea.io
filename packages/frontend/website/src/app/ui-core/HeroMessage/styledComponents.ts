import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
