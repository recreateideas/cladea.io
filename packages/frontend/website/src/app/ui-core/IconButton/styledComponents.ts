import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
