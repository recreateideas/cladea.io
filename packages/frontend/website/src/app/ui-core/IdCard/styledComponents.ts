import styled, { css } from "styled-components";
import { avatar } from "../images";

export const Container = styled.div<{ breakpoint: string }>`
  ${(props) => {
    const {
      theme: {
        dsl: { layout, palette },
      },
      breakpoint,
    } = props;
    const isSmall = breakpoint === "sm";
    const width = isSmall ? "100%" : "30vw";
    return css`
      border-radius: 12px;
      min-width: ${width};
      background-color: ${palette.neutrals.quaternaryBg};
      border: solid 1px ${palette.neutrals.borders};
      overflow: hidden;
      .background {
        width: 100%;
        height: 60vh;
        fill: ${palette.neutrals.secondaryBg};
      }
      z-index: ${layout.namedZIndex.leftSidebar - 1};
      .bio {
        display: flex;
        flex-direction: column;
      }
      .avatar {
        position: relative;
        border-radius: 100%;
        overflow: hidden;
        background: ${palette.neutrals.primaryBg};
        width: 20vh;
        height: 20vh;
        margin: 24px auto;
        .image-container {
          border-radius: 100%;
          overflow: hidden;
          width: inherit;
          height: inherit;
          background-image: url(${avatar});
          background-position: center;
          background-repeat: no-repeat;
          background-size: cover;
        }
        .space {
          background: unset !important;
        }
      }
      .name {
        display: flex;
        letter-spacing: 1px;
        justify-content: center;
        gap: 16px;
        .first {
        }
        .last {
          color: ${palette.tertiary.pink[500]};
        }
      }
      .position {
        width: 100%;
        .title {
          color: ${palette.secondary.neon[500]};
          text-align: center;
        }
      }
      .contacts {
        display: flex;
        flex-direction: column;
        gap: 16px;
        margin: ${isSmall ? "2vh" : "24px"} 24px;
        padding: ${isSmall ? "2vh" : "24px"};
        border: solid 1px ${palette.neutrals.borders};
        border-radius: 8px;
        background-color: ${palette.neutrals.secondaryBg};
        .contact-container {
          width: 100%;
          /* height: 40px; */
          display: flex;
          justify-content: center;
          .href-container {
            text-decoration: none;
            display: flex;
            gap: 16px;
            color: ${palette.fonts.primary};
            cursor: pointer;
            img {
              width: 36px;
            }
            .href-text {
              display: flex;
              flex-direction: column;
              justify-content: center;
              width: 80px;
            }
          }
        }
      }
    `;
  }};
`;

export const ToastContainer = styled.div`
  ${(props) => {
    const {
      theme: {
        dsl: { palette },
      },
    } = props;
    return css`
      .toast {
        display: block;
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translateX(-50%);
        min-width: 450px;
        background-color: ${palette.neutrals.quaternaryBg};
        box-shadow: ${palette.neutrals.boxShadow(palette.secondary.neon[500])};
        border: solid 1px ${palette.neutrals.borders};
        .btn-close {
          color: ${palette.fonts.primary};
          background-color: ${palette.secondary.neon[500]};
        }
        .toast-header {
          justify-content: space-between;
          background-color: ${palette.neutrals.primaryBg};
          button {
            margin-right: 0;
          }
          .header-title {
            font-weight: 600;
            width: 100%;
            color: ${palette.secondary.neon[500]};
          }
        }

        .body-content {
          width: 100%;
          display: flex;
          justify-content: space-between;
          .copied-value {
            color: ${palette.tertiary.pink[500]};
          }
        }
      }
    `;
  }};
`;
