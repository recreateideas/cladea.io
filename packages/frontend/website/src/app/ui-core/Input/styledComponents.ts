import styled from 'styled-components';
import { CustomCss } from '@types';

export const Container = styled.div<{ customCss?: CustomCss }>`
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${(props) => props.customCss}
`;
