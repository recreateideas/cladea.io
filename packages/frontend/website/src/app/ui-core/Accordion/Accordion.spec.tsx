import { Accordion } from './Accordion';
import { Accordion as StyledAccordion } from './styledComponents';
import { TestProvider } from 'src/tests';

const { mountWithProviderAndTheme, shallow } = TestProvider();
describe('<Accordion/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Accordion />, state);
        expect(root).toMatchSnapshot();
    });
    it('should be closed by default', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Accordion />, state);
        const collapse = root.find('.accordion-details').at(0);
        expect(collapse.length).toBe(0);
    });
    it('should respect the defaultOpen prop', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Accordion defaultExpanded={true} />, state);
        const collapse = root.find('.accordion-details').at(0);
        expect(collapse.length).toBe(1);
    });
    it('should open on accordion change', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Accordion />, state);
        const accordion = root.find(StyledAccordion).at(0);
        accordion.props().onChange();
        root.update();
        const collapse = root.find('.accordion-details').at(0);
        expect(collapse.length).toBe(1);
        // expect(root).toMatchSnapshot();
    });
    it('should open when its controlled and expanded prop changes to true', () => {
        const root = shallow(
            <Accordion controlled={true}>
                <div className="children" />
            </Accordion>
        );
        let collapse = root.find('.children').at(0);
        root.setProps({ expanded: true });
        root.update();
        collapse = root.find('.children').at(0);
        expect(collapse.length).toBe(1);
    });
});
