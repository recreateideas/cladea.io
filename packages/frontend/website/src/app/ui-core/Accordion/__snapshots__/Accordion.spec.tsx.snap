// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> should render 1`] = `
Array [
  .c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c0 {
  background-color: #fff;
  color: #031A51;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.c2 {
  position: relative;
  -webkit-transition: margin 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: margin 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  overflow-anchor: none;
  border-radius: 0;
}

.c2:before {
  position: absolute;
  left: 0;
  top: -1px;
  right: 0;
  height: 1px;
  content: "";
  opacity: 1;
  background-color: rgba(0,0,0,0.12);
  -webkit-transition: opacity 150ms cubic-bezier(0.4,0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4,0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c2:first-of-type:before {
  display: none;
}

.c2.Mui-expanded:before {
  opacity: 0;
}

.c2.Mui-expanded:first-of-type {
  margin-top: 0;
}

.c2.Mui-expanded:last-of-type {
  margin-bottom: 0;
}

.c2.Mui-expanded + .c1.Mui-expanded:before {
  display: none;
}

.c2.Mui-disabled {
  background-color: rgba(0,0,0,0.12);
}

.c2:first-of-type {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c2:last-of-type {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c2.Mui-expanded {
  margin: 16px 0;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4::-moz-focus-inner {
  border-style: none;
}

.c4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 48px;
  padding: 0px 16px;
  -webkit-transition: min-height 150ms cubic-bezier(0.4,0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: min-height 150ms cubic-bezier(0.4,0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c5.Mui-focusVisible {
  background-color: rgba(0,0,0,0.12);
}

.c5.Mui-disabled {
  opacity: 0.38;
}

.c5:hover:not(.Mui-disabled) {
  cursor: pointer;
}

.c5.Mui-expanded {
  min-height: 64px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 12px 0;
  -webkit-transition: margin 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: margin 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c7.Mui-expanded {
  margin: 20px 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: rgba(0,0,0,0.54);
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition: transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c8.Mui-expanded {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c3 {
  background-color: #f9fafb;
  box-shadow: none;
  background-image: none;
}

.c6 {
  padding: 0px 8px;
}

.c6.Mui-expanded {
  min-height: 48px;
}

.c6 > div {
  margin: 8px 0 !important;
}

@supports (-ms-ime-align:auto) {
  .c2:last-of-type {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
  }
}

@media print {
  .c4 {
    color-adjust: exact;
  }
}

<div
    className=""
  >
    <div
      className="c0 MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c1 c2 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters c3"
    >
      <div
        aria-expanded={false}
        className="c4 MuiButtonBase-root c5 MuiAccordionSummary-root MuiAccordionSummary-gutters c6 accordion-summary"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="c7 MuiAccordionSummary-content MuiAccordionSummary-contentGutters"
        >
          <div
            className=""
          />
        </div>
        <div
          className="c8 MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "",
]
`;
